mlrun/feature_store/steps.py

```diff
@@ -226,34 +227,132 @@ def _do_pandas(self, event):
                     min_val = val_range[0] if val_range[0] != "-inf" else -np.inf
                     max_val = val_range[1] if val_range[1] != "inf" else np.inf
                     otherwise = ""
-                    if new_column_name in event.columns:
-                        otherwise = event[new_column_name]
-                    event = event.withColumn(
+                    if new_column_name in df.columns:
+                        otherwise = df[new_column_name]
+                    df = df.withColumn(
                         new_column_name,
                         when(
-                            (event[column] < max_val) & (event[column] >= min_val),
+                            (df[column] < max_val) & (df[column] >= min_val),
                             lit(val),
                         ).otherwise(otherwise),
                     )
 
         if not self.with_original_features:
-            event = event.select(*self.mapping.keys())
+            df = df.select(*self.mapping.keys())
 
-        return event
+        return df
+
+    @classmethod
+    def validate_args(cls, feature_set, **kwargs):
+        mapping = kwargs.get("mapping", [])
+        for column, column_map in mapping.items():
+            is_range = False
+            if not cls.get_ranges_key() in column_map:
+                types = set(
+                    type(val)
+                    for val in column_map.values()
+                    if type(val) is not None
+                    and not (
+                        isinstance(val, (float, np.float64, np.float32, np.float16))
+                        and math.isnan(val)
+                    )
+                )
+            else:
+                if len(column_map) > 1:
+                    raise mlrun.errors.MLRunInvalidArgumentError(
+                        f"MapValues - mapping values of the same column can not combine ranges and single replacement"
+                        f"- '{column}'"
+                    )
+                is_range = True
+                ranges_dict = column_map[cls.get_ranges_key()]
+                types = set()
+                for ranges_mapping_values in ranges_dict.values():
+                    range_types = set(
+                        type(val)
+                        for val in ranges_mapping_values
+                        if type(val) is not None
+                        and val != "-inf"
+                        and val != "inf"
+                        and not (
+                            isinstance(val, (float, np.float64, np.float32, np.float16))
+                            and math.isnan(val)
+                        )
+                    )
+                    types = types.union(range_types)
+            if len(types) > 1:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"MapValues - mapping {'range ' if is_range else ''}values of the same column must be in the"
+                    f" same type! Column - '{column}'"
```

Comment:
```suggestion
                    f"MapValues - mapping values of the same column must of the same type, which was not the case for "
                    f"column '{column}'"
```
The user can figure out if it's range or not from the column, if it matters to them.

Reply:
fixed